test/regions.cpp

```diff
@@ -250,3 +250,39 @@ TEST(Regions, Docs_4)
         ASSERT_EQ(gold[i], output[i])<<" mismatch at i="<<i<<std::endl;
     }
 }
+
+TEST(Regions, Issue1650)
+{
+    // input data
+    char input[64] =  {
+        1, 1, 1, 1, 1, 1, 1, 1,
```

Comment:
You should probably make a test that would test the multiple block scenario. You can make a vector like this `vector<char> input(64 * 64, 1);` instead of setting the values for each element.

Reply:
By multiple block scenario, do you many components ?

Reply:
I think I got what you meant, dims spanning multiple blocks.